openingd/openingd.c

```diff
@@ -825,6 +825,7 @@ static u8 *funder_channel(struct state *state,
 fail:
 	if (taken(utxos))
 		tal_free(utxos);
+	tal_free(state);
```

Comment:
I'm not sure this is a good idea: `state` is allocated/owned by `main()`, where it'll also be freed on exit. If we `tal_free` it here we create a double-free bug. In addition this is specular to `fundee_channel` which does not free `state`. Maybe it's better to just mark `state` as a root element using `notleak`.